File: MyApplication10/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView resulttv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);



        setContentView(R.layout.activity_main);

        resulttv=findViewById(R.id.result);

         




    }
    public void onDigitclick(View v){
          Button button=(Button)v;
          resulttv.append(button.getText());


    }

    String operator="";
    String LHS="";
   public void clickedonoperator(View v){
        Button button=(Button) v;
        if(operator.isEmpty()){
            operator=button.getText().toString();
            LHS= resulttv.getText().toString();
           resulttv.setText(null);
            Log.e("MainActivity","operatoe  "+operator+", LHS: "+LHS );


        }else{
            String RHS=resulttv.getText().toString();
        double res=calculate(LHS,operator,RHS);
            Log.e("Main-clickedonoperator","result  "+res);
            LHS=res+"";
           operator=button.getText().toString();
            resulttv.setText("");


        }





   }
   public double calculate(String LHS,String operator,String RHS){
          Double lhs=new Double(LHS);
          Double rhs=new Double(RHS);
          if(operator.equals("+")){
              return lhs+rhs;

          }else if(operator.equals("-")){
              return lhs-rhs;

          }else if(operator.equals("*")){
              return lhs*rhs;

          }else{
              return lhs/rhs;

          }



   }
   public void onequal(View v){
      String RHS=resulttv.getText().toString();
      double res=calculate(LHS,operator,RHS);
      resulttv.setText(res+"");
       operator="";
       LHS="";


   }


}

File: MyApplication/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Gravity
import android.view.View
import android.widget.ArrayAdapter
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    val fruit= arrayOf("apple","bananna","peach","watermelon","lemons","apple","bananna","peach","watermelon","lemons")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val myadapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, fruit)
        mylist.adapter = myadapter
        mylist.setOnItemClickListener { _, view, position, l ->
            val textview = view as TextView
            when (position) {
                0 -> {
                    Toast.makeText(this, textview.text.toString(), Toast.LENGTH_LONG).show()
                }
                1 -> {
                    Toast.makeText(this, textview.text.toString(), Toast.LENGTH_LONG).show()
                }
                2 -> {
                    Toast.makeText(this, textview.text.toString(), Toast.LENGTH_LONG).show()
                }
                3 -> {
                    Toast.makeText(this, textview.text.toString(), Toast.LENGTH_LONG).show()
                }

            }


        }




        fun toaste(s: String) {
            val context = applicationContext
            val duration = Toast.LENGTH_LONG
            val toast = Toast.makeText(context, s, duration)
            toast.show()
            toast.setGravity(Gravity.CENTER, 0, 0)


        }
    }
}
